OpenOversight/app/__init__.py

```diff
@@ -77,21 +77,33 @@ def create_app(config_name="default"):
     # Also log when endpoints are getting hit hard
     limiter.logger.addHandler(file_handler)
 
-    @app.errorhandler(404)
-    def page_not_found(e):
-        return render_template("404.html"), 404
-
-    @app.errorhandler(403)
-    def forbidden(e):
-        return render_template("403.html"), 403
-
-    @app.errorhandler(500)
-    def internal_error(e):
-        return render_template("500.html"), 500
-
-    @app.errorhandler(429)
-    def rate_exceeded(e):
-        return render_template("429.html"), 429
+    # Define error handlers
+    def create_errorhandler(code, error, template):
+        """
+        Create an error handler that returns a JSON or a template response
+        based on the request "Accept" header.
+        :param code: status code to handle
+        :param error: response error message, if JSON
+        :param template: template response
+        """
+
+        def _handler_method(e):
+            if request.accept_mimetypes.best == "application/json":
+                return jsonify(error=error), code
+            return render_template(template), code
+
+        return _handler_method
+
+    error_handlers = [
+        (403, "Forbidden", "403.html"),
```

Comment:
I'd prefer that we use constants for the response codes: https://docs.python.org/3/library/http.html

We could also then use f-strings for the third value in the tuple.

Reply:
Would you be alright if I made these changes on the branch, @abandoned-prototype? I could also submit this in a different PR if you'd like.

Reply:
I would prefer a follow up (meaning a separate PR). I am actually working on one myself right now and I have included this particular suggestion. If you want to make a chance based on any of your other comments, I would appreciate if you open a PR for that. Thanks!

Reply:
Issue made: https://github.com/lucyparsons/OpenOversight/issues/909